src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/TagHelperSemanticRangeVisitor.cs

```diff
@@ -188,6 +188,41 @@ public override void VisitCSharpStatementBody(CSharpStatementBodySyntax node)
         }
     }
 
+    public override void VisitCSharpImplicitExpressionBody(CSharpImplicitExpressionBodySyntax node)
+    {
+        // Generally same as explicit expression, below, but different because the parens might not be there,
+        // and because the compiler isn't nice and doesn't give us OpenParen and CloseParen properties we can
+        // easily use.
+
+        // Matches @(SomeCSharpCode())
+        if (node.CSharpCode.Children is
+            [
+                CSharpExpressionLiteralSyntax { LiteralTokens: [{ Kind: SyntaxKind.LeftParenthesis } openParen] },
+                CSharpExpressionLiteralSyntax body,
+                CSharpExpressionLiteralSyntax { LiteralTokens: [{ Kind: SyntaxKind.RightParenthesis } closeParen] },
```

Comment:
@333fred I don't know if you have a list of "things the syntax tree could do better/more consistently", but this is one. If the node were a `CSharp*Ex*plicitExpressionBody` I could just use the `OpenParen` and `CloseParen` properties and not need the pattern match.